src/App.tsx

```diff
@@ -1,55 +1,93 @@
-import React from 'react';
+import React, {
+  useState, useMemo, useCallback,
+} from 'react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './PeopleList';
+
+function debounce(callback: (...args: any[]) => void, delay: number) {
+  let timerId = 0;
+
+  return (...args: any[]) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const delay = 1000;
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [dropdownFocused, setDropdownFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [delay],
+  );
+
+  const handleSelectedPerson = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setAppliedQuery('');
+    setDropdownFocused(false);
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+
+    if (event.target.value !== appliedQuery) {
+      applyQuery(event.target.value);
+      setAppliedQuery(event.target.value);
+    }
+  };
+
+  const handleFocus = () => {
+    if (selectedPerson) {
+      setQuery('');
+    }
+
+    setDropdownFocused(true);
+  };
+
+  const handleBlur = () => {
+    setTimeout(() => {
+      setDropdownFocused(false);
+    }, 200);
+  };
```

Comment:
```suggestion
  const handleBlur = () => {
      setDropdownFocused(false);
  };
```
we can avoid timeout when will use in PeopleInfo `onMouseDown` event instead `onClick` event.

Reply:
I apologize, there was indeed an error with deploy. Thank you for pointing it out!